MoviesTVSentiments/app/src/main/java/com/google/moviestvsentiments/service/account/AccountDao.java

```diff
@@ -0,0 +1,42 @@
+package com.google.moviestvsentiments.service.account;
+
+import androidx.room.Dao;
+import androidx.room.Query;
+import com.google.moviestvsentiments.model.Account;
+import java.util.List;
+
+/**
+ * AccountDao provides a high-level interface for querying the accounts database table.
+ */
+@Dao
+public interface AccountDao {
+
+    /**
+     * addAccount adds the given account name into the accounts table. If the name already exists,
+     * it is ignored. The account record is created with is_current set to false.
+     * @param name The name of the account to add.
+     */
+    @Query("INSERT OR IGNORE INTO accounts_table (account_name) VALUES (:name)")
+    public void addAccount(String name);
+
+    /**
+     * getAlphabetizedAccounts returns a list of all accounts in alphabetical order.
+     */
+    @Query("SELECT * FROM accounts_table ORDER BY account_name ASC")
+    public List<Account> getAlphabetizedAccounts();
+
+    /**
+     * @return The currently in-use account.
+     */
+    @Query("SELECT * FROM accounts_table WHERE is_current = 1")
+    public Account getCurrentAccount();
+
+    /**
+     * Updates the given account record's is_current value.
+     * @param name The name of the account to update.
+     * @param isCurrent The value to set is_current to.
+     * @return The number of updated records. This will be 1 if the account exists and 0 otherwise.
```

Comment:
Since name is a parameter, is it possible to have more than one update? If the update can only be 0 or 1, would boolean be more logical?

Reply:
I changed it to a boolean, but I'm not sure what you're asking in the first question. I was planning on passing (name, true) to this method when someone selects an account to sign in with. Then when they log out, I pass (name, false) to the method.

Reply:
What I was asking is the possibility of multiple DB updates when name is the primary key of the account table and also the parameter of the function here. Basically, this function should produce at most one update right?

Reply:
Yes, that's correct.